src/features/Hook/useFacilities.tsx

```diff
@@ -0,0 +1,30 @@
+import { useState, useEffect, useContext } from "react";
+import axios from "axios";
+import { faciRoomsUrl } from "../../services/api";
+import { AuthContext } from "../../context/AuthContext";
+
+const useFacilities = () => {
+    const { requestHeaders } = useContext(AuthContext);
+    const [facilitiesList, setFacilitiesList] = useState([]);
+
+    const getAllFacilities = () => {
+        axios
+            .get(`${faciRoomsUrl}`, {
+                headers: requestHeaders,
+            })
+            .then((response) => {
+                setFacilitiesList(response?.data?.data?.facilities);
+
+            })
+            .catch((error) => {
+
+            });
+    };
+    useEffect(() => {
+        getAllFacilities();
+      }, [requestHeaders]);
+
+      return { facilitiesList };
```

Comment:
This is currently great, but you'll need later to return a function that will handle the pagination when this hook is used in the facilities page.
You can also return the loading state of the request, as we usually need it to show a loading spinner for the user.